File: frontend/types/Task.ts
export interface TaskSource {
  uuid: string;
  source_type: string;
  source_url?: string;
  source_id?: string;
  title?: string;
  content?: string;
  extra_data?: string;
  created_at?: number;
  updated_at?: number;
}

export interface Task {
  uuid: string;
  title: string;
  description?: string;
  completed: boolean;
  expires_at?: number;
  sources?: TaskSource[];
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  expires_at?: number;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string;
  completed?: boolean;
  expires_at?: number;
}

// ソースタイプ関連の型定義
export interface SourceTypeConfig {
  icon: React.ComponentType<{ className?: string }>;
  displayName: string;
  color?: string;
}

export type SourceTypeKey =
  | "email"
  | "calendar"
  | "slack"
  | "teams"
  | "discord"
  | "github_issue"
  | "github_pr"
  | "jira"
  | "trello"
  | "asana"
  | "notion"
  | "linear"
  | "clickup"
  | "other";

File: backend/app/models/task.py
from uuid import UUID

from app.schema import SourceType, TaskPriority
from pydantic import BaseModel


class TaskSourceModel(BaseModel):
    model_config = {"from_attributes": True}

    uuid: UUID | None = None
    source_type: SourceType
    source_url: str | None = None
    source_id: str | None = None
    title: str | None = None
    content: str | None = None
    extra_data: str | None = None
    created_at: float | None = None
    updated_at: float | None = None


class TaskModel(BaseModel):
    model_config = {"from_attributes": True}

    uuid: UUID | None = None
    title: str
    description: str | None = None
    completed: bool = False
    expires_at: float | None = None
    priority: TaskPriority | None = None
    created_at: float | None = None
    updated_at: float | None = None
    sources: list[TaskSourceModel] = []


class CreateTaskRequest(BaseModel):
    title: str
    description: str | None = None
    expires_at: float | None = None
    priority: TaskPriority | None = None


class UpdateTaskRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    expires_at: float | None = None
    completed: bool | None = None
    priority: TaskPriority | None = None
